### ISSUE TITLE
Replace usage of has_key() in examples

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6485]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#6485|
|Type|enhancement|
|Created|2013-04-27 18:48:53Z|

The following documentation example files contain calls to the deprecated `x.has_key(y)` method. They should be replaced with `y in x` instead.

```
./doc/core/howto/tutorial/listings/finger/finger_config.py:24:    if config.has_key('ircnick'):
./doc/core/howto/tutorial/listings/finger/finger_config.py:32:    if config.has_key('pbport'):
./doc/core/howto/tutorial/listings/finger/finger/finger.py:354:    if config.has_key('ircnick'):
./doc/core/howto/tutorial/listings/finger/finger/finger.py:362:    if config.has_key('pbport'):
./doc/core/howto/listings/servers/chat.py:16:        if self.users.has_key(self.name):
./doc/core/howto/listings/servers/chat.py:26:        if self.users.has_key(name):
./doc/core/howto/listings/pb/chatserver.py:17:        if not self.groups.has_key(groupname):
```

This is part of [#4053](https://github.com/twisted/twisted/issues/4053).

Attachments:

* [ticket6485.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/6bf/6bff223c6454901a4414857a7304f43fa28f5e67/7193d2d608c3d84825f12423dfb6b8872b883e7e.patch) (3539 bytes) - added by canibanoglu on 2013-04-30 18:50:27Z - Updated the dictionary key lookup syntax, not so sure about the top file though
* [ticket6485.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/6bf/6bff223c6454901a4414857a7304f43fa28f5e67/43d677a53bf3c58e92dff6e43a68d9c0929c9e46.patch) (3339 bytes) - added by jackson15j on 2013-07-06 10:12:48Z - Fixed up the top file from original patch
* [ticket6485.3.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/6bf/6bff223c6454901a4414857a7304f43fa28f5e67/acdff2a009487acff538a3c31b36a00da350a513.patch) (3129 bytes) - added by jackson15j on 2013-07-06 10:36:16Z - 2nd attempt at fixing topfile. Now .misc and blank.

<details><summary>Searchable metadata</summary>

```
trac-id__6485 6485
type__enhancement enhancement
reporter__thijs thijs
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1367088533000000 1367088533000000
changetime__1373108455000000 1373108455000000
version__None None
owner__exarkun exarkun
cc__hynek
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="canibanoglu's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|canibanoglu set owner to canibanoglu|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="canibanoglu's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|canibanoglu removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_3"></a>|@hynek commented|
|-|-|

Hi canibanoglu,

the code LGTM, however your topfile  needs indeed some love.

The word “syntax” is spelled wrong at the first occurrence and there’s no need to list all the files that have been changed. Make a general one-line statement about example Python files instead please. For further details have a look at http://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles .

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_4"></a>|@Julian commented|
|-|-|

It also probably is just a misc (empty) topfile, I think doc topfiles are generally for whole new or large changes to documentation.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @exarkun|
|-|-|

Looks good, thanks!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [39079]) Apply ticket6485.3.patch - replace `dict.has_key(key)` with `key in dict` in some core examples.

Author: canibanoglu, jackson15j
Reviewer: hynek, Julian, exarkun
Fixes: [#6485](https://github.com/twisted/twisted/issues/6485)
